### PR TITLE
IGNITE-20226 Sql. Fix muted tests after disallowed char -> varbinary casting

### PR DESCRIPTION
**Pull request by zstan**
…casting

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

**Comment by zstan** [comment]
@lowka can you make a review plz ?

**Review by lowka — APPROVED** [review]

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/uuid/ItUuidDmlTest.java [hunk 1]
```diff
@@ -17,16 +17,78 @@
...
 
 /**
  * Tests for DML statements for {@link UuidType UUID data type}.
  */
 public class ItUuidDmlTest extends BaseDmlDataTypeTest<UUID> {
+    /** {@code INSERT} with dynamic parameters is not allowed for compatible types. */
+    @ParameterizedTest
+    @MethodSource("convertedFrom")
+    public void testInsertFromDynamicParameterFromConvertible(TestTypeArguments arguments) {
+        var t = assertThrows(IgniteException.class, () -> {
```
Comment (ygerzhedovich) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/uuid/ItUuidDmlTest.java:43]: please use `org.apache.ignite.internal.sql.engine.util.SqlTestUtils#assertThrowsSqlException`

Comment (ygerzhedovich) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/uuid/ItUuidDmlTest.java:43]: like this
`
        assertThrowsSqlException(
                Sql.STMT_VALIDATION_ERR,
                "Values passed to VALUES operator must have compatible types",
                () -> runSql("INSERT INTO t VALUES (1, ?)", arguments.argValue(0))
        );
`

Comment (zstan) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/uuid/ItUuidDmlTest.java]: @ygerzhedovich fixed but this activity beyond the scope of this issue.

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/uuid/ItUuidExpressionTest.java [hunk 2]
```diff
@@ -65,6 +70,31 @@ public void testInvalidUuidStringInDynamicParams() {
         assertThat(t.getMessage(), containsString("Invalid UUID string"));
     }
 
+    /**
+     * {@code COALESCE} operator does not allow different types in its arguments.
+     */
+    @ParameterizedTest
+    @MethodSource("convertedFrom")
+    public void testCoalesceMissingTypesIsIllegal(TestTypeArguments arguments) {
+        IgniteException t = assertThrows(IgniteException.class, () -> {
```
Comment (ygerzhedovich) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/uuid/ItUuidExpressionTest.java:79]: let's fix all places. It should be also 'assertThrowsSqlException' with error code check

**Comment by zstan** [comment]
@ygerzhedovich done )

**Review by ygerzhedovich — APPROVED** [review]

**Pull request merged**